### DISCUSSION TITLE
怎样编译DDE？

### DISCUSSION THREAD

**Discussion by xrui94** [linuxdeepin/developer-center#3973] [Development Questions | 开发答疑]
我想修改DDE的UI，但是我应该怎样编译DDE呢？这里有两个问题：
（1）可以看到Git的仓库里有很多DDE的项目，但是不知道哪个项目是入口项目？
（2）如果我修改了源码，并成功编译好了新的DDE，那我应该如何安装呢？我的预期是，编译后，可以重新登录系统，然后选择我自己修改过的DDE。

**Comment by BLumia** [comment]
> 可以看到Git的仓库里有很多DDE的项目，但是不知道哪个项目是入口项目？

DDE 的组件很多，不同组件归属于不同项目。需要改哪个取决于你希望修改哪个组件。

> 如果我修改了源码，并成功编译好了新的DDE，那我应该如何安装呢？

同样取决于你改的是哪个组件，但大部分都是 Qt CMake 项目，构建完成后 `make install` （或者对应的 install target）即可。不过还是建议具体项目分别看待。

顺便，如果你对参与开发感兴趣的话，可以考虑加入我们的开发者 [Matrix](https://wiki.deepin.org/Matrix) 或 Telegram 群 :)
